### DISCUSSION TITLE
Handling files (or arrays of events) as data source in the streaming environments

### DISCUSSION THREAD

**Discussion by AkthemRehab** [apache/streampipes#2299] [Q&A]
Hello,
I want to ingest files (containing arrays of events like time waveform vibration readings) to Streampipes, forward the files as an http post to an api, and then collect the response from the api and present it in the dashboard/data explorer. 

Further, I would also benefit from the ability to storing the file in Streampipes and showing the time wave form, FFT, and other signal processing values in the Streampipes dashboard and data explorer.

Thanks,
Akthem

**Comment by tenthe** [comment]
Hi @AkthemRehab,
thanks for the discussion. 

I have seen that we also have an adapter that launches a REST endpoint to post events to. However, I am not quite sure how this adapter works.
Maybe you can take a look at it, it is called `HTTP Server`. 
Do you think you can use this adapter for your use case?

@obermeier you mentioned the adapter in #2199. 
Since I could not find any documentation for the adapter in [1], would it be possible for you to briefly explain how it works. Thank you very much! 

Cheers,
Philipp
[1] https://github.com/apache/streampipes/tree/dev/streampipes-extensions/streampipes-connect-adapters-iiot/src/main/resources/org.apache.streampipes.connect.iiot.protocol.stream.httpserver

**Reply by obermeier** [reply]
Hi @tenthe, thank you for the hint. I will check the documentation.

@AkthemRehab: The ```HTTP Server``` creates a resource at ```/endpoints/{endpointName}``` where you can post events to. e.g.  in JSON format.

![grafik](https://github.com/apache/streampipes/assets/1632575/b0001785-cbe1-454e-bc53-832a6ab30094)

Don't hesitate to ask if you have further questions.

Cheers,
Stefan

**Reply by obermeier** [reply]
The described feature is not available in release 0.93.0 (Related issue #2199).
You can use version 0.92.0, older versions, or fix the issue manually in the latest versions.

If you have any questions on how to do it, don't hesitate to ask
